### PR TITLE
LPS-164433 | Add test JSURLCanBeRepeated and RemoteAppsRepeatableFields.testcase

### PR DESCRIPTION
**Pull request by bronsonsoda**
**Background**
Create remote test that ensures multiple urls can be added

**Test plan**
_JSURLCanBeRepeated_
Given Remote Application > Custom Elements create page
When click repeatable button for JS URL
Then new JS URL field is added
And Then repeatable button + and - are present for both JS URL fields

https://issues.liferay.com/browse/LPS-164433

### CODE REVIEW

**Comment by bronsonsoda** [comment]
ci:test:frontend:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			RemoteApps.addCustomElement(
+				entryHtmlName = "${customElementHtmlName}",
+				entryName = "${customElementName}",
+				entryURL = "${customElementURL}");
+		}
+
+		task ("When click repeatable button for JS URL") {
+			RemoteAppsEntry.addCSSURLRow();
+
+			RemoteAppsEntry.addCSSURLRow();
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:60]: addCSSURLRow is not the same as JS URL. We're looking for `RemoteAppsEntry.addURLRow();` also, we're only supposed to click 1 time. Multiple times should be covered in different testcase https://issues.liferay.com/browse/LPS-164435

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		}
+
+		task ("Then new JS URL field is added") {
+			AssertElementPresent(locator1 = "RemoteAppsEntry#URL_ROW_INDEX");
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:64]: no `key_index` or `key_id` parameters defined for locator.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		}
+
+		task ("And Then repeatable button + and - are present for both JS URL fields") {
+			AssertElementPresent(locator1 = "RemoteAppsEntry#ADD_URL_ROW");
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:68]: You need both + and -. This xpath only has `add-row` which doesn't gaurantee checking - is also present.

**Review by john-co — CHANGES_REQUESTED** [review]

**Comment by john-co** [comment]
@bronsonsoda please make requested changes and make sure test passes when ready for review.

**Comment by bronsonsoda** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+					layoutName = "${layoutName}");
+
+				Navigator.gotoPage(pageName = "Home");
+
+				PortalInstances.tearDownCP();
+			}
+		}
+	}
+
+	@description = "This is a test for LPS-164433. The JSN URL can be repeatably clicked."
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:39]: `The JSON* URL can be`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		property portal.acceptance = "true";
+
+		task ("Given Remote Application > Custom Elements create page") {
+			RemoteApps.goToRemoteAppsPortlet();
+
+			RemoteApps.addCustomElement(
+				entryHtmlName = "vanilla-counter",
+				entryName = "Vanilla Counter",
+				entryURL = "http://remote-component-test.wincent.com/packages/vanilla-counter/index.js");
+		}
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:51]: We don't need to add a custom element, we just need to navigate to the "create page" meaning click on plus button and click on Custom Element.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+		task ("When click repeatable button for JS URL") {
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:53]: This task is just for clicking the repeatable button, not publishing or typing anything in it. Does this make sense?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+		task ("Then new JS URL field is added") {
+			RemoteAppsEntry.clickRemoteEntryName(remote_name = "Vanilla Counter");
+
+			AssertElementPresent(
+				key_url = "http://www.localhost.com",
+				locator1 = "RemoteAppsEntry#REMOTE_URL_FIELD");
+		}
+
+		task ("And Then repeatable button + and - are present for both JS URL fields") {
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteAppsRepeatableFields.testcase:75]: We need to validate 2 sets of + and - `both JS URL fields`.  This means we need xpaths for the first URL field and 2nd URL field. + and -. So 4 paths that could be simplified to 2 paths with `key_number` or something corresponding to each URL field

**Review by john-co — CHANGES_REQUESTED** [review]
Hi @bronsonsoda , please review and update. Let me know if have any questions.

**Comment by bronsonsoda** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
ci:test:sf

**Review by john-co — APPROVED** [review]
Please review changes.

**Comment by john-co** [comment]
ci:forward

**Pull request closed without merging**